langlib/lang.string/src/main/java/org/ballerinalang/langlib/string/PadZero.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.langlib.string;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BString;
+
+import static io.ballerina.runtime.api.constants.RuntimeConstants.STRING_LANG_LIB;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.LENGTH_SHOULD_BE_LESS_THAN_2147483647;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.getModulePrefixedReason;
+
+/**
+ * Extern function ballerina.model.strings:PadZero.
+ *
+ * @since 2201.1.0
+ */
+
+public class PadZero {
+
+    public static BString padZero(BString str, long len, BString padChar) {
+        int strLength = str.length();
+        if (len <= strLength) {
+            return str;
+        }
+
+        if (len > Integer.MAX_VALUE) {
+            throw ErrorCreator.createError(getModulePrefixedReason(STRING_LANG_LIB,
+                    LENGTH_SHOULD_BE_LESS_THAN_2147483647));
+        }
+
+        int targetLen = (int) len;
+        String pad = padChar.toString().repeat(targetLen - strLength);
+        String result;
+        if (str.substring(0, 1).toString().equals("+") || str.substring(0, 1).toString().equals("-")) {
+            result = str.substring(0, 1) + pad + str.substring(1, strLength);
```

Comment:
Shall we use a joiner instead of + here? Also please check which one ( using +, joiner or stringBuilders) is more efficient here.

Reply:
Also without `substring`ing you can check the index

Reply:
Also adding  `0` after `+` or `-` is useful if the string is a number right? what will happen if there are multiple signs together? does the rule still valid? (`The zeros are added at the start of the string, after a `+` or `-` sign if there is one.`)

Reply:
According to the [spec discussion](https://github.com/ballerina-platform/ballerina-spec/issues/770#issuecomment-791258822), 
```
Padding would be done by providing three functions in lang.string with equivalent functionality to Python ljust, rjust and zfill.
```

here `zfill` is equivalent to ballerina `padZero()`. Python defines `zfill` as,

```
Return a copy of the string left filled with ASCII '0' digits to make a string of length width. A leading sign prefix ('+'/'-') is handled by inserting the padding after the sign character rather than before. The original string is returned if width is less than or equal to len(s).
```
In Python, they have clearly mentioned that `A leading sign prefix ('+'/'-') is handled by inserting the padding after the sign character rather than before`. 
Since we are implementing the string padding with equivalent functionality to Python, the current behavior is okay compared to Python.
BTW we can ask to modify the definition of padZero into something like `The zeros are added at the start of the string, after the leading + or - sign if there is one.`?

Reply:
Example for zfill(),
<img width="166" alt="image" src="https://user-images.githubusercontent.com/34838308/160746718-9855a9b4-a786-4694-9acf-cae9c0cae525.png">